Sprint 17

desikvasya opened the pull request:
Реализовано контекстное меню для треков на главном экране;
Есть скриншот-тесты для главного экрана;
Реализована тёмная тема для главного экрана;
Подключена аналитика AppMetrica с помощью CocoaPods и присылаются события для действий на главном экране;
Реализован экран статистики;
Есть английская локализация на главном экране;
Проведено код-ревью в группах.

desikvasya committed:
Добавил локализацию

Надпись с количеством дней на трекере теперь отображается корректно

desikvasya committed:
Локализация приложения
- Английский язык

desikvasya committed:
Переведен счетчик дней

desikvasya committed:
Добавил контекстное меню
Теперь можно удалить трекер

desikvasya committed:
Работаю над закреплением трекера

desikvasya committed:
Скриншот-тесты

desikvasya committed:
Темная тема для главного экрана
Скриншот-тесты прошли успешно

desikvasya committed:
Цвет плюса на главном экране

desikvasya committed:
Редактирование привычек и событий

desikvasya committed:
Добавил статистику

desikvasya committed:
Интегрировал метрику

desikvasya committed:
Навел порядок

So1ovyov commented on Habit/TrackerCreation/NewIrregularEventViewController.swift:
```diff
@@ -288,13 +354,13 @@ extension NewIrregularEventViewController: UICollectionViewDataSource {
                 fatalError("Unable to dequeue CollectionHeaderSupplementaryView")
             }
-            header.title.text = "Цвет"
+            header.title.text = NSLocalizedString("NewIrregularEventViewController.colorHeader", comment: "")
```
Здесь и дальше можно попробовать другой хинт, например
```
title.text  = "NewIrregularEventViewController.colorHeader".localized(withComment: ""
```
Подробнее https://stackoverflow.com/questions/25081757/whats-nslocalizedstring-equivalent-in-swift

So1ovyov commented on Habit/TrackerCreation/NewIrregularEventViewController.swift:
```diff
@@ -184,6 +197,31 @@ final class NewIrregularEventViewController: UIViewController {
+    
+    
     // MARK: - Методы, вызываемые при нажатии кнопок
     // MARK: Метод, вызываемый при нажатии на кнопку "Отмена"
```
Сам раздел с одинаковыми типами сущностей можно пометить //MARK: - Private methods, а вот сами методы можно документировать через ///

So1ovyov commented on Habit/TrackerCreation/NewIrregularEventViewController.swift:
```diff
@@ -166,6 +175,10 @@ final class NewIrregularEventViewController: UIViewController {
             secondStack.centerXAnchor.constraint(equalTo: scroll.centerXAnchor),
             secondStack.topAnchor.constraint(equalTo: colorCollection.bottomAnchor, constant: 24),
         ])
+        //        if eventToEdit != nil {
```
Кажется нужно удалить

So1ovyov commented on Habit/TabBarControllers/TrackersViewController.swift:
```diff
@@ -374,3 +398,125 @@ extension TrackersViewController {
+                }
+            }
+            
+            let editAction = UIAction(title: NSLocalizedString("Touch.edit", comment: ""), image: nil) {
```
Для лучшей читабельности можно попробовать 
```
let editAction = UIAction(
       title: NSLocalizedString("Touch.edit", comment: ""), 
       image: nil
) { [weak self] _ in
```

So1ovyov commented on Habit/TabBarControllers/TrackersViewController.swift:
```diff
@@ -374,3 +398,125 @@ extension TrackersViewController {
     }
 }
+
+// MARK: - Расширение для UIContextMenuInteractionDelegate
```
В разработке прийнято подписывать расширения просто // MARK: - UIContextMenuInteractionDelegate, без надписи расширение/extension

So1ovyov commented on Habit/YandexMetrica/AnalyticService.swift:
```diff
@@ -0,0 +1,20 @@
+
+final class AnalyticsService {
+    
+    func report(event: String, params: [AnyHashable : Any]) {
```
Кажется не хватает логики

desikvasya committed:
Исправления после код-ревью
- добавил аналитику для кнопки плюс в ячейке с трекером
- исправил комментарии
- небольшие изменения для повышения читаемости
